Gems/Atom/Feature/Common/Code/Source/FrameCaptureSystemComponent.cpp

```diff
@@ -389,94 +389,80 @@ namespace AZ
         }
 
-        FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshotForWindow(const AZStd::string& filePath, AzFramework::NativeWindowHandle windowHandle)
+        FrameCaptureSystemComponent::CaptureHandle FrameCaptureSystemComponent::ScreenshotPreparation(
+            const AZStd::string& imagePath, AZ::RPI::AttachmentReadback::CallbackFunction callbackFunction)
         {
             if (!CanCapture())
             {
-                return InvalidFrameCaptureId;
+                return CaptureHandle::Null();
             }
 
             CaptureHandle captureHandle = InitCapture();
             if (captureHandle.IsNull())
             {
-                AZ_Assert(false, "Failed to allocate a capture");
-                return InvalidFrameCaptureId;
+                AZ_Error("FrameCaptureSystemComponent", false, "Failed to allocate a capture.");
+                return CaptureHandle::Null();
             }
+
             AZStd::scoped_lock<CaptureHandle> scope_lock(captureHandle);
             CaptureState* capture = captureHandle.GetCaptureState();
             if (!capture) // failed to get the capture state ptr, abort
             {
                 m_idleCaptures.push_back(captureHandle);
-                return InvalidFrameCaptureId;
+                AZ_Error("FrameCaptureSystemComponent", false, "Failed to get the captureState.");
+                return CaptureHandle::Null();
             }
 
-            // Find SwapChainPass for the window handle
-            RPI::SwapChainPass* pass = AZ::RPI::PassSystemInterface::Get()->FindSwapChainPass(windowHandle);
-            if (!pass)
+            if (!capture->m_readback->IsReady())
             {
-                AZ_Warning("FrameCaptureSystemComponent", false, "Failed to find SwapChainPass for the window");
                 m_idleCaptures.push_back(captureHandle);
-                return InvalidFrameCaptureId;
+                AZ_Error("FrameCaptureSystemComponent", false, "Failed to capture attachment since the readback is not ready.");
+                return CaptureHandle::Null();
             }
 
-            if (!capture->m_readback->IsReady())
+            capture->m_readback->SetUserIdentifier(captureHandle.GetCaptureStateIndex());
+            if (callbackFunction != nullptr)
             {
-                AZ_Assert(false, "Failed to capture attachment since the readback is not ready");
-                m_idleCaptures.push_back(captureHandle);
-                return InvalidFrameCaptureId;
+                capture->m_readback->SetCallback(callbackFunction);
+            }
+            else
+            {
+                capture->m_readback->SetCallback(
+                    AZStd::bind(&FrameCaptureSystemComponent::CaptureAttachmentCallback, this, AZStd::placeholders::_1));
+            }
+            if (!imagePath.empty())
+            {
+                capture->m_outputFilePath = ResolvePath(imagePath);
             }
```

Comment:
We don't support supplying an output filepath if a custom callback function is supplied, so this block should probably be moved up inside the `callbackFunction == nullptr` block. And possibly an assert added to the `callbackFunction != nullptr` block